CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/ReplicatorConfigurationRequestHandler.java

```diff
@@ -0,0 +1,260 @@
+        Authenticator authenticator = args.get("authenticator");
+        // ConflictResolver conflictResolver = args.get("conflictResolver");
+        Map<String, String> headers = args.get("headers");
+
+        if (replicatorType == null)
+        {
+            replicatorType = "push_pull";
+        }
+        replicatorType = replicatorType.toLowerCase();
+        ReplicatorConfiguration.ReplicatorType replType;
+        if (replicatorType.equals("push")) {
+            replType = ReplicatorConfiguration.ReplicatorType.PUSH;
+        } else if (replicatorType.equals("pull")) {
+            replType = ReplicatorConfiguration.ReplicatorType.PULL;
+        } else {
+            replType = ReplicatorConfiguration.ReplicatorType.PUSH_AND_PULL;
+        }
+        ReplicatorConfiguration config;
+        if (sourceDb != null && targetURL != null) {
+            URLEndpoint target = new URLEndpoint(targetURL);
+            config = new ReplicatorConfiguration(sourceDb, target);
+        } else if (sourceDb != null && targetDb != null) {
+            DatabaseEndpoint target = new DatabaseEndpoint(targetDb);
+            config = new ReplicatorConfiguration(sourceDb, target);
+        } else {
+            throw new Exception("\"No source db provided or target url provided\"");
+        }
+        if (continuous != null) {
+            config.setContinuous(continuous);
+        }
+        else {
+            config.setContinuous(false);
+        }
+        if (headers != null) {
+            config.setHeaders(headers);
+        }
+        config.setAuthenticator(authenticator);
+        config.setReplicatorType(replType);
+        /*if (conflictResolver != null) {
+            config.setConflictResolver(conflictResolver);
+        }*/
+        if (channels != null) {
+            config.setChannels(channels);
+        }
+        if (documentIds != null) {
+            config.setDocumentIDs(documentIds);
+        }
+        System.out.println(args);
+        if (pinnedservercert != null){
+            Context context = com.couchbase.CouchbaseLiteServ.MainActivity.getAppContext();
+            byte[] ServerCert = this.getPinnedCertFile(context);
+            // Set pinned certificate.
+            config.setPinnedServerCertificate(ServerCert);
+        }
+        return config;
+    }
+
+    public ReplicatorConfiguration create(Args args) {
+        ReplicatorConfiguration config = args.get("configuration");
+        return config;
+    }
+
+    public Authenticator getAuthenticator(Args args){
+        ReplicatorConfiguration replicatorConfiguration = args.get("configuration");
+        return replicatorConfiguration.getAuthenticator();
+    }
+
+    public List<String>  getChannels(Args args){
+        ReplicatorConfiguration replicatorConfiguration = args.get("configuration");
+        return replicatorConfiguration.getChannels();
+    }
+
+    /*public ConflictResolver getConflictResolver(Args args){
```

Comment:
Can this commented code be removed?

Reply:
will wait for 6 months and remove, if dev get custom conflict resolver later , this might be needed